### PR TITLE
Example Integration with ZIO ZLayer

### PR DESCRIPTION
**Pull request by tampler**

### CODE REVIEW

File: examples/src/main/scala/sttp/tapir/examples/LayerEndpoint.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  object UserService {
+    trait Service {
+      def hello(id: Int): ZIO[Any, String, Pet]
+    }
+
+    val live: ZLayer[Console, Nothing, Has[Service]] = ZLayer.fromFunction { console: Console =>
+      new Service {
+        def hello(id: Int): ZIO[Any, String, Pet] = {
+          console.get.putStrLn(s"Got Pet request for $id")
```
Comment (adamw) [inline examples/src/main/scala/sttp/tapir/examples/LayerEndpoint.scala:18]: shouldn't this be combined with the next effect using `>>`?

File: examples/src/main/scala/sttp/tapir/examples/ZioExampleHttp4sServer.scala [hunk 2]
```diff
@@ -34,13 +32,16 @@ object ZioExampleHttp4sServer extends App {
   val petEndpoint: Endpoint[Int, String, Pet, Nothing] =
     endpoint.get.in("pet" / path[Int]("petId")).errorOut(stringBody).out(jsonBody[Pet])
 
+  val zioEndpoint: Endpoint[Int, String, Pet, Nothing] =
```
Comment (adamw) [inline examples/src/main/scala/sttp/tapir/examples/ZioExampleHttp4sServer.scala:35]: maybe we can have just a single endpoint, that is, convert `petEndpoint` to the "ZIO-layer" version? Having two different ones might be confusing to users.

Or maybe, a comment on two possible approaches would be better?

File: playground/src/main/scala/sttp/tapir/example/zio/LayerEndpoint.scala [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package sttp.tapir.examples
```
Comment (adamw) [inline playground/src/main/scala/sttp/tapir/example/zio/LayerEndpoint.scala:1]: the examples project is where examples should live, playground is just for quick experimentation :)

File: examples/src/main/scala/sttp/tapir/examples/ZioExampleHttp4sServer.scala (hunk 2 above)
Comment (tampler) [reply examples/src/main/scala/sttp/tapir/examples/ZioExampleHttp4sServer.scala:35]: Adam I didn't understand your point.
I have 2 EP there - "petEndpoint" and "zioEndpoint". And combine them in a service.

If you don't like this, make your choice and give me an exaple
Thanks for reviewing this

File: playground/src/main/scala/sttp/tapir/example/zio/LayerEndpoint.scala (hunk 3 above)
Comment (tampler) [reply playground/src/main/scala/sttp/tapir/example/zio/LayerEndpoint.scala:1]: Got your message. So what to do  ?

Comment (adamw) [reply playground/src/main/scala/sttp/tapir/example/zio/LayerEndpoint.scala:1]: I'd just leave the example in the `example` module and remove from `playground` :)

File: examples/src/main/scala/sttp/tapir/examples/ZioExampleHttp4sServer.scala (hunk 2 above)
Comment (adamw) [reply examples/src/main/scala/sttp/tapir/examples/ZioExampleHttp4sServer.scala:35]: I guess my suggestion is to add a comment, which clarifies what's the difference in the implementation of the `petEndpoint` and `zioEndpoint`

File: playground/src/main/scala/sttp/tapir/example/zio/LayerEndpoint.scala (hunk 3 above)
Comment (tampler) [reply playground/src/main/scala/sttp/tapir/example/zio/LayerEndpoint.scala:1]: Done

File: examples/src/main/scala/sttp/tapir/examples/ZioExampleHttp4sServer.scala (hunk 2 above)
Comment (tampler) [reply examples/src/main/scala/sttp/tapir/examples/ZioExampleHttp4sServer.scala]: Updated

**Comment by adamw** [comment]
Thanks!

**Comment by tampler** [comment]
Seems like there was some compile problems, no ?

**Comment by adamw** [comment]
Yes, fixing:)

**Comment by tampler** [comment]
Ah okay, Adam. Thanks for your support. Will make another MR cleaner

**Pull request merged**